logrotate.c

```diff
@@ -2523,6 +2524,14 @@ int main(int argc, const char **argv)
     logSetLevel(MESS_NORMAL);
     setlocale (LC_ALL, "");
 
+    if (stat(STATEDIR, &st) == -1) {
+        mkdir(STATEDIR, 0755);
+    }
+
+    if (stat(OLDSTATEFILE, &st) == 0) {
+        unlink(OLDSTATEFILE);
```

Comment:
I do not think that existing state file should be silently removed.  We should move it to the new location instead.